document/core/exec/instructions.rst

```diff
@@ -614,6 +614,385 @@ Memory Instructions
+
+6. Assert: due to :ref:`validation <valid-atomic-rmwn>`, a value of :ref:`value type <syntax-valtype>` :math:`t` is on the top of the stack.
+
+7. Pop the value :math:`t.\CONST~c2` from the stack.
+
+8. Assert: due to :ref:`validation <valid-atomic-rmwn>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+
+9. Pop the value :math:`\I32.\CONST~i` from the stack.
+
+10. Let :math:`\X{ea}` be :math:`i + \memarg.\OFFSET`.
+
+11. If :math:`N` is not part of the instruction, then:
+
+    a. Let :math:`N` be the :ref:`bit width <syntax-valtype>` :math:`|t|` of :ref:`value type <syntax-valtype>` :math:`t`.
+
+12. If :math:`\X{ea} + N/8` is larger than the length of :math:`\X{mem}.\MIDATA`, then:
+
+    a. Trap.
+
+13. If :math:`\X{ea}` modulo :math:`N/8` is not equal to :math:`0`, then:
+
+    a. Trap.
+
+14. Let :math:`b^\ast_r` be the byte sequence :math:`\X{mem}.\MIDATA[\X{ea} \slice N/8]`.
+
+15. Let :math:`c1` be the integer for which :math:`bytes_{\iN}(m) = b^\ast_r`.
+
+16. Let :math:`m` be the result of computing :math:`\atomicop_t(c1, c2)`.
+
+17. If :math:`N` is part of the instruction, then:
+
+    a. Let :math:`n` be the result of computing :math:`\wrap_{|t|,N}(m)`.
+
+    b. Let :math:`b^\ast_w` be the byte sequence :math:`\bytes_{\iN}(n)`.
+
+18. Else:
+
+    a. Let :math:`b^\ast_w` be the byte sequence :math:`\bytes_t(m)`.
+
+19. Replace the bytes :math:`\X{mem}.\MIDATA[\X{ea} \slice N/8]` with :math:`b^\ast_w`.
+
+20. Push the value :math:`t.\CONST~c1` to the stack.
+
+.. math::
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   S; F; (\I32.\CONST~i)~(t.\CONST~c2)~(t.\ATOMICRMW.\atomicop~\memarg) &\stepto& S'; F; (t.\CONST~c1)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+     (\mbox{if} & \X{ea} = i + \memarg.\OFFSET \\
+     \wedge & F.\AMODULE.\MIMEMS[0] = a \\
+     \wedge & \X{ea} + |t|/8 \leq |S.\SMEMS[a].\MIDATA| \\
+     \wedge & \X{ea} \mod |t|/8 = 0 \\
+     \wedge & \bytes_t(c1) = S.\SMEMS[a].\MIDATA[\X{ea} \slice |t|/8]) \\
+     \wedge & m = \atomicop_t(c1, c2) \\
+     \wedge & S' = S \with \SMEMS[a].\MIDATA[\X{ea}:|t|/8] = \bytes_t(m)
+     \end{array} \\
+   \begin{array}{lcl@{\qquad}l}
+   S; F; (\I32.\CONST~i)~(t.\CONST~c2)~(t.\ATOMICRMW{N}\K{\_u}.\atomicop~\memarg) &\stepto& S'; F; (t.\CONST~c1)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+     (\mbox{if} & \X{ea} = i + \memarg.\OFFSET \\
+     \wedge & F.\AMODULE.\MIMEMS[0] = a \\
+     \wedge & \X{ea} + N/8 \leq |S.\SMEMS[a].\MIDATA| \\
+     \wedge & \X{ea} \mod N/8 = 0 \\
+     \wedge & \bytes_t(c1) = S.\SMEMS[a].\MIDATA[\X{ea} \slice N/8]) \\
+     \wedge & m = \atomicop_t(c1, c2) \\
+     \wedge & S' = S \with \SMEMS[a].\MIDATA[\X{ea}:N/8] = \bytes_{\iN}(\wrap_{|t|,N}(m))
+     \end{array} \\
+   \begin{array}{lcl@{\qquad}l}
+   S; F; (\I32.\CONST~k)~(t.\CONST~c)~(t.\ATOMICRMW({N}\K{\_u})^?.\atomicop~\memarg) &\stepto& S; F; \TRAP
```

Comment:
The hyperlink on atomicop doesn't seem to work (missing link target on the syntax/instructions page?).